### ISSUE TITLE
Loading all nuclides into materials

### ISSUE THREAD

**Issue by pshriwise** [pyne/pyne#1075]
[This line](https://github.com/pyne/pyne/blob/dc750cad9cf4f2e938e46a8d2920390417951307/src/material.cpp#L150) seems to load all nuclides as part of the material composition, even if they aren't part of that material. Is there a reason for this I'm unaware of?

**Comment by pshriwise** [comment]
I'll close this. I ended up tracking it down. The HDF5 readers expect there to be contributions from each nuclide in the file. Most material formatting functions check the contribution value before writing it out (like [this](https://github.com/pyne/pyne/blob/develop/src/material.cpp#L659)). I'll do the same in the `openmc` writer soon. 

In case this comes in handy later, I did notice interesting [piece of code](https://github.com/pyne/pyne/blob/dc750cad9cf4f2e938e46a8d2920390417951307/src/material.cpp#L289) where the writer will use the existing number of nuclides to set the total number to write to if a material library is already in the file. I'm not sure what would happen if the new material had more nuclides than were in the previous material.
